app/validators/previously_published_validator.rb

```diff
@@ -0,0 +1,16 @@
+class PreviouslyPublishedValidator < ActiveModel::Validator
+  def validate(record)
+    record.has_previously_published_error = false
+
+    if record.previously_published.nil?
+      record.errors[:base] << "You must specify whether the document has been published before"
+      record.has_previously_published_error = true
+    elsif record.previously_published && record.first_published_at.blank?
+      record.errors.add(:first_published_at, "can't be blank")
+    end
+
+    if record.first_published_at && Time.zone.now < record.first_published_at
+      record.errors.add(:first_published_at, "can't be set to a future date")
+    end
```

Comment:
Sorry - I'm sure you don't need another comment from me on this long running thing 😞 

Since this is pretty distinct from previously_published do we just want this as a separated validation?

Reply:
Thanks! Addressed in f17bcab. Seemed small enough to have directly in the validations, but happy to extract it in a method if not.